aws-lc-rs/scripts/run-rustls-integration.sh

```diff
@@ -31,13 +37,34 @@ cargo download rustls | tar xvzf - -C "${RUSTLS_DIR}" --strip-components=1
 RUSTLS_COMMIT="$(jq -r '.git.sha1' ${RUSTLS_DIR}/.cargo_vcs_info.json)"
 rm -rf "${RUSTLS_DIR}" # Cleanup before we clone
 
+git clone https://github.com/rustls/rcgen "${RUSTLS_RCGEN_DIR}"
 git clone https://github.com/rustls/webpki.git "${RUSTLS_WEBPKI_DIR}"
 git clone https://github.com/rustls/rustls.git "${RUSTLS_DIR}"
 
+# Update rcgen to use the GitHub repository reference under test.
+pushd "${RUSTLS_RCGEN_DIR}"
+git checkout "${RUSTLS_RCGEN_COMMIT}"
+cargo add --path "${ROOT}/aws-lc-rs" --package rcgen
+cargo add --path "${ROOT}/aws-lc-rs" --package rustls-cert-gen
+cargo update
+cargo update "aws-lc-rs@${AWS_LC_RS_VERSION}"
+cargo test --features aws_lc_rs
+popd &>/dev/null # "${RUSTLS_RCGEN_DIR}"
+
 # Update rustls-webpki to use the GitHub repository reference under test.
 pushd "${RUSTLS_WEBPKI_DIR}"
 git checkout "${RUSTLS_WEBPKI_COMMIT}"
-cargo add --path "${ROOT}/aws-lc-rs"
+WEBPKI_RCGEN_STRING="^rcgen = { .* }"
+WEBPKI_RCGEN_PATH_STRING="rcgen = { path = \"${RUSTLS_RCGEN_DIR}/rcgen\" , default-features = false, features = [\"aws_lc_rs\"] }"
+WEBPKI_AWS_LC_RS_STRING="^aws-lc-rs = { version.* }"
+WEBPKI_AWS_LC_RS_PATH_STRING="aws-lc-rs = { path = \"${ROOT}/aws-lc-rs\", optional = true, default-features = false, features = [\"aws-lc-sys\"] }"
+if [[ "$(uname)" == "Darwin" ]]; then
+	find ./ -type f  -name "Cargo.toml" | xargs sed -i '' -e "s|${WEBPKI_RCGEN_STRING}|${WEBPKI_RCGEN_PATH_STRING}|g" -e "s|${WEBPKI_AWS_LC_RS_STRING}|${WEBPKI_AWS_LC_RS_PATH_STRING}|g"
+else
+	find ./ -type f  -name "Cargo.toml" | xargs sed -i -e "s|${WEBPKI_RCGEN_STRING}|${WEBPKI_RCGEN_PATH_STRING}|g" -e "s|${WEBPKI_AWS_LC_RS_STRING}|${WEBPKI_AWS_LC_RS_PATH_STRING}|g"
+fi
+# Trigger Cargo to generate the lock file
+cargo update
```

Comment:
Out of curiosity to change this from using cargo to using sed? Seems like the cargo route is less fragile to breakage with file format changes.

Reply:
cargo works fine when there's just one package in the workspace that needs a dependency update, but it seems to be unable to perform this operation when multiple packages need the update. I struggled to get cargo to do it for awhile, then gave up and switched to using sed to update the `Cargo.toml` files, which is what we also use for s2n-quic integration.